flink-formats/flink-csv/src/main/java/org/apache/flink/formats/csv/CsvFileFormatFactory.java

```diff
@@ -136,6 +147,45 @@ public BulkFormat<RowData, FileSourceSplit> createRuntimeDecoder(
         public ChangelogMode getChangelogMode() {
             return ChangelogMode.insertOnly();
         }
+
+        @Override
+        public TableStats reportStatistics(List<Path> files, DataType producedDataType) {
+            final int totalSampleLineCnt = 100;
+            try {
+                long totalSize = 0;
+                int sampledLineCnt = 0;
+                long sampledTotalSize = 0;
+                for (Path file : files) {
+                    FileSystem fs = FileSystem.get(file.toUri());
+                    FileStatus status = fs.getFileStatus(file);
+                    totalSize += status.getLen();
+
+                    // sample the line size
+                    if (sampledLineCnt < totalSampleLineCnt) {
+                        try (InputStreamReader isr =
+                                new InputStreamReader(
+                                        Files.newInputStream(new File(file.toUri()).toPath()))) {
+                            BufferedReader br = new BufferedReader(isr);
+                            String line;
+                            while (sampledLineCnt < totalSampleLineCnt
+                                    && (line = br.readLine()) != null) {
+                                sampledLineCnt += 1;
+                                sampledTotalSize += line.length();
+                            }
+                        }
+                    }
+                }
+                if (sampledTotalSize == 0) {
+                    return TableStats.UNKNOWN;
+                }
+
+                int realSampledLineCnt = Math.min(totalSampleLineCnt, sampledLineCnt);
+                int estimatedRowCount = (int) (totalSize * realSampledLineCnt / sampledTotalSize);
+                return new TableStats(estimatedRowCount);
```

Comment:
Why the estimation doesn't give the column stat?

Reply:
> Why the estimation doesn't give the column stat?

Not likes Orc and Parquet format, Csv format not support collect stats. The only way to collect column stats is to build a csv reader and read all files which is a heavy operation.